include/deal.II/differentiation/ad/ad_helpers.h

```diff
@@ -3280,6 +3280,379 @@ namespace Differentiation
+       * $\mathbf{g} \subset \boldsymbol{\Psi}(\mathbf{X})$
+       * for a subset of independent variables
+       * $\mathbf{A} \subset \mathbf{X}$, i.e.
+       * @f[
+       *   \mathbf{J}(\mathbf{g})
+       *      = \frac{\partial\mathbf{g}(\mathbf{X})}{\partial\mathbf{A}}
+       * @f]
+       *
+       * @param[in] jacobian The Jacobian of the vector function with respect to
+       * all independent variables, i.e., that returned by compute_jacobian().
+       * @param[in] extractor_row An extractor associated with the input field
+       * variables for which the first index of the Jacobian is extracted.
+       * This effectively defines which components of the global set
+       * of the dependent variables that this field (representing a subset of
+       * dependent functions) is associated with.
+       * @param[in] extractor_col An extractor associated with the input field
+       * variables for which the second index of the Jacobian is extracted.
+       * This effectively defines which components of the global set
+       * of independent variables that this field (representing a subset of the
+       * independent variables) is associated with.
+       */
+      template <typename ExtractorType_Row, typename ExtractorType_Col>
+      typename internal::VectorFieldJacobian<dim,
+                                             scalar_type,
+                                             ExtractorType_Row,
+                                             ExtractorType_Col>::type
+      extract_jacobian_component(const FullMatrix<scalar_type> &jacobian,
+                                 const ExtractorType_Row &      extractor_row,
+                                 const ExtractorType_Col &extractor_col) const;
+
+      /**
+       * Extract the Jacobian of the subset of dependent functions
+       * $\mathbf{g} \subset \boldsymbol{\Psi}(\mathbf{X})$
+       * for a subset of independent variables
+       * $\mathbf{A} \subset \mathbf{X}$, i.e.
+       * @f[
+       *   \mathbf{J}(\mathbf{g})
+       *      = \frac{\partial\mathbf{g}(\mathbf{X})}{\partial\mathbf{A}}
+       * @f]
+       *
+       * This function is a specialization of the above for rank-0 tensors
+       * (scalars).
+       */
+      Tensor<0, dim, scalar_type>
+      extract_jacobian_component(
+        const FullMatrix<scalar_type> &   jacobian,
+        const FEValuesExtractors::Scalar &extractor_row,
+        const FEValuesExtractors::Scalar &extractor_col) const;
+
+      /**
+       * Extract the Jacobian of the subset of dependent functions
+       * $\mathbf{g} \subset \boldsymbol{\Psi}(\mathbf{X})$
+       * for a subset of independent variables
+       * $\mathbf{A} \subset \mathbf{X}$, i.e.
+       * @f[
+       *   \mathbf{J}(\mathbf{g})
+       *      = \frac{\partial\mathbf{g}(\mathbf{X})}{\partial\mathbf{A}}
+       * @f]
+       *
+       * This function is a specialization of the above for rank-4 symmetric
+       * tensors.
```

Comment:
This corresponds to subsetting a whole set of rows and columns, and then reshaping the matrix so obtained into a `SymmetricTensor<4,dim>`. I like this operation. But can you say something along the lines of my first sentence? I think that would make it clearer.

Reply:
I've added this to the unspecialised version of the function. I think that its appropriate to put all of the details there (since its the most rigorously documented version of the function).